graphql_compiler/schema_transformation/merge_schemas.py

```diff
@@ -20,40 +42,58 @@
 )
 
 
-MergedSchemaDescriptor = namedtuple(
-    "MergedSchemaDescriptor",
-    (
-        "schema_ast",  # Document, AST representing the merged schema
-        "schema",  # GraphQLSchema, representing the same schema as schema_ast
-        "type_name_to_schema_id",
-        # Dict[str, str], mapping type name to the id of its schema, includes Interface, Object,
-        # Union, and Enum types
-    ),
-)
+@dataclass
+class MergedSchemaDescriptor:
+    """Describes a merged schema."""
 
+    # Both representing the merged schema.
+    schema_ast: DocumentNode
+    schema: GraphQLSchema
 
-CrossSchemaEdgeDescriptor = namedtuple(
-    "CrossSchemaEdgeDescriptor",
-    (
-        "edge_name",  # str, name used for the corresponding in and out fields
-        "outbound_field_reference",  # FieldReference namedtuple for the outbound field
-        "inbound_field_reference",  # FieldReference namedtuple for the inbound field
-        "out_edge_only",  # bool, whether or not the edge is bidirectional
-    ),
-)
+    # Mapping type name to the id of its schema. Includes interface, object, union, and enum types.
+    # Excludes scalars and directives because the same set of scalars and directives are expected to
+    # be defined in every schema.
+    type_name_to_schema_id: Dict[str, str]
 
 
-FieldReference = namedtuple(
-    "FieldReference",
-    (
-        "schema_id",  # str, identifier for the schema of the field
-        "type_name",  # str, name of the object or interface that the field belongs to
-        "field_name",  # str, name of the field, used in the stich directive
-    ),
-)
+@dataclass
+class FieldReference:
+    """Describes a particular field, including the type and schema to which it belongs."""
+
+    # Identifier for the schema to which this field belongs.
+    schema_id: str
+
+    # Name of object or interface to which this field belongs.
+    type_name: str
+
+    # Name of the field to be used in the @stitch directive.
+    field_name: str
+
+
+@dataclass
+class CrossSchemaEdgeDescriptor:
+    """Describes an edge that spans two schema."""
+
+    # Edge name excluding the "in_"/"out_" prefixes.
+    edge_name: str
+
+    # Describing the outbound field of the edge.
+    outbound_field_reference: FieldReference
 
+    # Describing the inbound field of the edge.
+    inbound_field_reference: FieldReference
 
-def merge_schemas(schema_id_to_ast, cross_schema_edges, type_equivalence_hints=None):
+    # Whether the edge is bidirectional.
+    out_edge_only: bool
+
+
+def merge_schemas(
+    schema_id_to_ast: "OrderedDict[str, DocumentNode]",
+    cross_schema_edges: List[CrossSchemaEdgeDescriptor],
+    type_equivalence_hints: Optional[
+        Dict[Union[GraphQLInterfaceType, GraphQLObjectType], GraphQLUnionType]
```

Comment:
`from ..schema.typedefs import TypeEquivalenceHintsType` might be useful here, it's just shorthand for the complex `Dict[...]` type signature.